app/src/main/java/ch/heigvd/pro/b04/android/Question/QuestionActivity.java

```diff
@@ -44,36 +48,24 @@ protected void onCreate(Bundle savedInstanceState) {
 
         answerList.setAdapter(answerAdapter);
         answerList.setLayoutManager(manager);
-/*
-        state.getSelectedAnswers().observe(this, answers -> {
-            for (Answer a: answers) {
-                question.answer(a);
-            }
-
-            if (question.getNumberOfAnswers() == answers.size()) {
-                question.answered();
-                System.out.println("All needed answers selected");
-            }
-
-            System.out.println("An Answer was selected");
-        });
-*/
     }
 
     public void goBack(View view) {
-        Intent intent = new Intent(this, QuestionActivity.class);
-        // TODO : get the previous question
+        Intent intent = new Intent(this, QuestionActivity.class)
+                .putExtra("question", state.getPreviousQuestion());
         startActivity(intent);
     }
 
     public void goNext(View view) {
-        Intent intent = new Intent(this, QuestionActivity.class);
-        // TODO : get the next question
+        Intent intent = new Intent(this, QuestionActivity.class)
+                .putExtra("question", state.getNextQuestion());
         startActivity(intent);
     }
 
     public void exitQuestion(View view) {
         Intent intent = new Intent(this, PollActivity.class);
+        intent.putExtra("idPoll", idPoll);
+        intent.putExtra("idModerator", idModerator);
         startActivity(intent);
     }
```

Comment:
Have you tried using `finish()` or something like that instead ? Maybe it's better ?